### PR TITLE
[MVT] VectorTile offline

### LINKED ISSUE #4612: [MVT] Offline save does not work
_OS/Device_
iOS Safari Chrome

_Steps to reproduce_
switch to BG swissimage or pixelkarte or vector tiles
try to save  extend

_result_

download stops at 0% for raster

_expected result_
Raster  offline works
vector offline does work s as well or show warning: " disabled in this version"

_Workaround:_
None

### PR DESCRIPTION
**Pull request by oterral**
Fix #4612

There is probably some bugs it s a first version.

<jenkins>[Test link](https://mf-geoadmin3.int.bgdi.ch/teo_offline/index.html)</jenkins>

### CODE REVIEW

**Comment by oterral** [comment]
@davidoesch if you can test a bit this one

**Comment by davidoesch** [comment]
if I'm offline I dont see the map offline button

reproduce:

- open link above
- switch to vec basemap
- save offline
- when offline complete switch to airplane mode
: result: 
- swissimage is a s backgound selected
- no offline perimeter

**Comment by davidoesch** [comment]
negative .... won't work https://photos.app.goo.gl/SbeMeTC8qQqD7K2x8

**Comment by oterral** [comment]
@davidoesch last try , I have created a branch with  the openlayers update : 
https://mf-geoadmin3.int.bgdi.ch/teo_offline_up/mobile.html?lang=en&topic=ech&bgLayer=ch.swisstopo.leichte-basiskarte.vt&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,&mobile=true

if it doesn't work at all on this branch . We should display an message that vector tiels are not saved.

**Comment by oterral** [comment]
It seems to work for me on chrome android.

**Comment by oterral** [comment]
@loicgasser can you have a try please?

**Comment by davidoesch** [comment]
it works -- when I go offline as long as the app is open and then go offline.
it does not work -- if start app from homescreen link. save map. close app. then go offline and restart app. ( it loads some 3D beta interface ...and yes i did clear cache..)

conclusio:
 enough invested here. it's ok for the test viewer

**Comment by loicgasser** [comment]
Save tiles works

But
I don't see the labels

When coming back on the page without internet, it seems to work.

When I come back online -> white map.
It basically breaks the app or at least the map.

**Comment by oterral** [comment]
ok I think I found the problem, when an  http request failed in offline that breaked the rendering of the app , can you please retry [here](https://mf-geoadmin3.int.bgdi.ch/teo_offline/index.html)  @davidoesch @loicgasser . This is my alst attempt :)

**Comment by davidoesch** [comment]
Naaw won't work on iOS . The rectangle is not shown during save and after it's finished. And in offline after a couple of zooms it crashes with offline https://photos.app.goo.gl/t2BhEcAdTPAYVhR47
Maybe this will be fixed/ related to your hack with cache=0 ?

**Comment by loicgasser** [comment]
@davidoesch this is a won't fix for the alpha. Should I deactivate the offline instead?

**Comment by davidoesch** [comment]
deactivate for alpha

File: Makefile [hunk 1]
```diff
@@ -188,7 +188,7 @@ NVM_DIR ?= $(HOME)/.nvm
 LESSC=source ${NVM_DIR}/nvm.sh && nvm use ${NODE_VERSION} && ${NODE_BIN}/lessc
 KARMA=source ${NVM_DIR}/nvm.sh && nvm use ${NODE_VERSION} && ${NODE_BIN}/karma
 PHANTOMJS=source ${NVM_DIR}/nvm.sh && nvm use ${NODE_VERSION} && ${NODE_BIN}/phantomjs
-NG_ANNOTATE=source ${NVM_DIR}/nvm.sh && nvm use ${NODE_VERSION} && ${NODE_BIN}/ng-annotate
+NG_ANNOTATE=${NODE_BIN}/ng-annotate
```
[the remark this answers is not in the document]
Comment (boecklic) [reply Makefile]: It didn't work...

**Comment by danduk82** [comment]
intergrated in `mvt_clean` but it is deactivated for the moment

**Pull request closed without merging**